contracts/crowdsale/validation/IndividuallyCappedCrowdsale.sol

```diff
@@ -0,0 +1,30 @@
+pragma solidity ^ 0.4.18;
+
+import "../../math/SafeMath.sol";
+import "../Crowdsale.sol";
+import "../../ownership/Ownable.sol";
+
+contract IndividuallyCappedCrowdsale is Crowdsale, Ownable {
+  using SafeMath for uint256;
+
+  mapping(address => uint256) public contributions;
+  mapping(address => uint256) public caps;
+
+  function setUserCap(address _beneficiary, uint256 _cap) external onlyOwner {
+    caps[_beneficiary] = _cap;
+  }
+
+  function getUserCap(address _beneficiary) public view returns (uint256) {
+    return caps[_beneficiary];
+  }
+
+  function getUserContribution(address _beneficiary) public view returns (uint256) {
+    return contributions[_beneficiary];
+  }
+
+  function _postValidatePurchase(address _beneficiary, uint256 _weiAmount) internal {
+    super._postValidatePurchase(_beneficiary, _weiAmount);
+    contributions[_beneficiary] = contributions[_beneficiary].add(_weiAmount);
```

Comment:
I don't like the fact that we are not only validating here, but also updating the user's contribution.
Perhaps we should add a new abstract method on the base Crowdsale to do this sort of thing?

Reply:
True, but the user contribution is only tracked for validation purposes.. We would need something like a separate `_updateValidationState()`.. Maybe too much, since we already have `_pre` and `_post` validation schemes.

Reply:
I agree that this new state member is only tracked for the sake of validation.
I also don't like _updateValidationState(), since as you say, seems a little to bloated.
Now, consider a crowdsale extension that simply keeps track of additional user data and allows users to query it individually or statistically. This function `processPurchase()` would be the entry point and the extension would not be a validation in itself. However, IndividuallyCappedCrowdsale as a validation would use `updatePurchase()`.

Reply:
Done. Please check whether we want it to be like this, as Crowdsale _is_ a bit more bloated now..